controlplane/api/v1alpha1/rke2controlplane_webhook.go

```diff
@@ -44,6 +47,12 @@ var _ webhook.Defaulter = &RKE2ControlPlane{}
 // Default implements webhook.Defaulter so a webhook will be registered for the type.
 func (r *RKE2ControlPlane) Default() {
 	bootstrapv1.DefaultRKE2ConfigSpec(&r.Spec.RKE2ConfigSpec)
+
+	if r.Spec.RegistrationMethod == RegistrationMethodAddress {
+		if r.Spec.ServerConfig.AdvertiseAddress == "" {
+			r.Spec.ServerConfig.AdvertiseAddress = r.Spec.RegistrationAddress
```

Comment:
can we log this properly communicate what is happening under the hood?

Reply:
Added a log line.